### DISCUSSION TITLE
RC Receive differences with rc-switch

### DISCUSSION THREAD

**Discussion by jab42** [arendst/Tasmota#21738] [Q&A]
Hi,
i see differences between Tasmota and rc-switch in receiving.
I.e.:
Tasmota: "RfReceived":{"Data":"0x18320","Bits":17,"Protocol":22,"Pulse":585}}

rc-lib: Decimal: 31967 (17Bit) Binary: 00111110011011111 Tri-State: not applicable PulseLength: 326 microseconds Protocol: 10
Raw data: 5881,1046,1153,310,1153,310,424,1038,426,1037,426,1037,428,1035,425,1040,1157,307,1152,311,426,1036,425,1033,1158,305,423,1039,424,1039,426,1040,425,1038,428,

It is an 1byOne Doorbell, so Protocol10 should be right.
How may i see, which one is correct?

**Comment by sfromis** [comment]
If you think that protocol 10 is "more correct" than protocol 22, you could try using the `RfProtocol` command to disable the latter.
https://tasmota.github.io/docs/RF-Protocol/#receiving-rf-codes

**Comment by jab42** [comment]
Thats not what i mean.
If i just enable protocol 10 it does not receive anything at all.

**Comment by jab42** [comment]
I found that the protocols differ between the two.
Protocol 10 at rc-switch lib is protocol 32 at tasmota.

Next Problem.
With the original sender tasmota receives:
{"Data":"0x18320","Bits":17,"Protocol":22,"Pulse":589}
but if i send with tasmota: {"Data":"0x7cdf","Bits":17,"Protocol":22,"Pulse":589}
Tasmota receives:
{"Data":"0x7cdf","Bits":17,"Protocol":2,"Pulse":589}
the original receiver doesnt detect that.
if i send
{"Data":"0x7cdf","Bits":17,"Protocol":32,"Pulse":362}
the original does nothing too.

what may i try to amalyze why the original doesnt detect my sends?
